day1/floor.py

```diff
@@ -2,11 +2,34 @@ class Floor:
     MOVE_UP = '('
     MOVE_DOWN = ')'
 
-    def __init__(self, input):
+    def __init__(self, input, basement):
         self.input = input
-    
+        self.basement = basement
+
     @property
-    def santa_current_floor(self):
+    def current_floor(self):
         moves_up = self.input.count(self.MOVE_UP)
         moves_down = self.input.count(self.MOVE_DOWN)
         return moves_up - moves_down
+
+    @property
+    def first_enter_basement(self):
+        first_enter_basement = 0
+        floor = 0
+        for move in self.input:
```

Comment:
Você consegue pegar o `index` aqui desse `for`?

Talvez algo como:

```python
for index, move in self.input:
```

🤔 

Porque se conseguir, vc não precisa definir a variável `first_enter_basement` e nem dar um `+= 1` nela, igual logo ali em baixo.

Daí poderia ficar algo do tipo, ali em baixo:

```python
if floor == BASEMENT:
    return index + 1
```

Talvez também seja mais performático. :)